File: Java/Mobile/src/main/java/org/tektutor/Mobile.java
package org.tektutor;

public class Mobile {

	private ICamera camera;

	public Mobile( ICamera camera ) {
		this.camera = camera;
	}

	public boolean powerOn() {

		if ( camera.on() ) {
			System.out.println ("Mobile - powerOn positive code path ...");
			return true;
		}
		
		System.out.println ( "Mobile - powerOn negative code path...");
		return false;
	}

}

File: C#/Mobile/MobileLib/Mobile.cs
using System;

namespace TekTutor
{
    public class Mobile
    {
        private ICamera camera;

        public Mobile(ICamera camera) //Constructor Dependency Injection
        {
            this.camera = camera;  //Aggregation - Shared ownership
        }

        public bool powerOn()
        {
            if ( camera.on() ) //dependency
            {
                //camera.on();
                Console.WriteLine("Mobile - powerOn Positive code path ...");
                return true;
            }


            Console.WriteLine("Mobile - powerOn Negative code path ...");
            return false;
        }
    }
}
